### DISCUSSION TITLE
RCM and connected component subgraphs and collection of cyles

### DISCUSSION THREAD

**Discussion by vsotofrances** [ZigRazor/CXXGraph#441] [Ideas]
I would find it helpful to have the following:
reverse Cuthill-Mckee ordering for a general graph RCM
In doing so, the subgraphs of connected components can be easily obtained.
Finally, the collection of graph cycles would be beneficial to me.

I am new to this Lib and do not know if there is a way to obtain some of these results.

P.S. Adding to the above....  does this function allow creating a subgraph from a subset of nodes?
`
//VIC
template <typename T>
const T_EdgeSet<T> &Graph<T>::getEdgeSet(std::unordered_set<std::string> NodeUserIdSet) const {
  CXXGraph::T_EdgeSet<T> NewEdgeSet;
  auto laplacianMatrix = this->getLaplacianMatrix();
  for(const auto &NodeUserId:NodeUserIdSet)
  {
    if(const auto Node=this->getNode(NodeUserId))
    {
      auto Neighbors=(*laplacianMatrix)[*Node] ;
      for(const auto &neighbor:Neighbors)// neighbor = (node,edge)
      {
        auto it=std::find(NodeUserIdSet.begin(),NodeUserIdSet.end(),neighbor.first->getUserId());
        if(neighbor.first != Node && it!=NodeUserIdSet.end())
        {
          NewEdgeSet.insert(neighbor.second);
        }
      }
    }
  }
  return NewEdgeSet;
}
// /VIC
`

**Comment by ZigRazor** [comment]
the Cuthill-Mckee algorithm is not implemented yet, you can open an issue for this.
For the connected components we have some algorithms like Tarjan, you can read more on DOCS or README.
For the Cycles you can check in the same way the DOCS or README

**Comment by vsotofrances** [comment]
Thanks a lot for your response, indeed.
Best regards
Victor
